src/ui/FacetRange/FacetRange.ts

```diff
@@ -117,14 +117,17 @@ export class FacetRange extends Facet implements IComponentBindings {
   }
 
   public getValueCaption(facetValue: FacetValue): string {
-    const lookupValueIsSpecified = !isUndefined(facetValue.lookupValue) && facetValue.lookupValue !== facetValue.value;
-    if (this.options.valueCaption || lookupValueIsSpecified) {
+    if (this.options.valueCaption || this.isLabelSpecifiedForValue(facetValue)) {
       return super.getValueCaption(facetValue);
     }
 
     return this.translateValuesFromFormat(facetValue);
   }
 
+  private isLabelSpecifiedForValue(facetValue: FacetValue) {
+    return this.options.ranges && this.options.ranges.filter(range => !isUndefined(range) && range.label === facetValue.lookupValue).length;
+  }
```

Comment:
It doesn't matter much, but I personally prefer using `indexOf` than `filter` when checking if any element matches a condition, since `indexOf` will stop searching at the first truthy encounter.

E.g.,
```ts
private isLabelSpecifiedForValue(facetValue: FacetValue) {
    return this.options.ranges && this.options.ranges.indexOf(range => !isUndefined(range) && range.label === facetValue.lookupValue) !== -1;
}
```

Reply:
Good point!

Reply:
Why is the `!isUndefined(range)` check needed? If we are looping through an array, aren't the elements defined?